File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  let obj={}
  for(let i=0;i<nums.length;i++){
      let temp=[]
      if(!obj[nums[i]]){
          temp.push(i)
          obj[nums[i]]=temp
      }else{
          obj[nums[i]].push(i)
      }
  }
    let j=0
    let k=nums.length-1
    nums.sort((num1,num2)=>num1-num2)
    let store=[]
    while(j<k){
        if(nums[j]+nums[k]===target){
            store.push(nums[j])
            store.push(nums[k])
            break
        }else if(nums[j]+nums[k]<target) j++
        else k--
    }
    let res=[]
    for(let i=0;i<2;i++){
        let idx=obj[store[i]].pop()
        res.push(idx)
    }
    return res
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, List<Integer>> obj = new HashMap<>();

        for (int k = 0; k < nums.length; k++) {
            if (!obj.containsKey(nums[k])) {
                obj.put(nums[k], new ArrayList<>());
            }
            obj.get(nums[k]).add(k);
        }

        Arrays.sort(nums);

        List<Integer> store = new ArrayList<>();
        int i = 0;
        int j = nums.length - 1;

        while (i < j) {
            if (nums[i] + nums[j] == target) {
                store.add(nums[i]);
                store.add(nums[j]);
                break;
            } else if (nums[i] + nums[j] < target) {
                i++;
            } else {
                j--;
            }
        }

        int[] res = new int[2];
        for (int k = 0; k < 2; k++) {
            res[k] = obj.get(store.get(k)).remove(obj.get(store.get(k)).size() - 1);
        }

        return res;
    }
}

